[USING] Chapter on Using CoinJoin (#342)
zkSNACKs/WasabiDoc: docs/using-wasabi/CoinJoin.md

---
{
  "title": "CoinJoin",
  "description": "A step by step guide on how to do Wasabi CoinJoins. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop."
}
---

# CoinJoin

> A [CoinJoin](https://en.bitcoin.it/Privacy#CoinJoin) is a special Bitcoin transactions where several peers get together to literally join their coins.

Reviewer: ```suggestion
A [CoinJoin](https://en.bitcoin.it/Privacy#CoinJoin) is a special Bitcoin transaction where several peers get together to literally join their coins in a single transaction.
```
Note: Many Bitcoin users are not native English speakers and are not tech savvies so I still think we should use `people` instead of `peers` to make it easy for readers to understand.
Same applies for other terms whenever possible

> They collaboratively build a transaction where each of them provides some coins in the input, and each of them fresh addresses in the output.

Reviewer: ```suggestion
They collaboratively build a transaction where each of them provides some coins as inputs, and fresh addresses as outputs.
```

If after a timeout not all outputs are registered, then this round is abandoned, the missing peers are banned, and a new round is started.

### Signing

Now that all inputs and outputs are registered, the Wasabi coordinator can start the [signing phase](https://github.com/nopara73/zerolink#3-signing-phase) by building the CoinJoin transaction with all the registered inputs, the anonset outputs, the change outputs and the coordinator fee output.
He sends this transaction to all the Alices of this round.
Each Alice verifies that:

* The committed round hash is equal to the hash of all the inputs in the proposed transaction.
* Her inputs and outputs are correctly included.

Then she signs the transaction with the private keys of her inputs.
Alice sends the uniqueID, the signature and the input index to the coordinator, who then verifies this information.

The signing phase ends when the coordinator has all the valid signatures for all the registered inputs.

### Broadcasting

The CoinJoin transaction is successfully built and signed, and it is now ready to be [broadcasted](https://github.com/nopara73/zerolink#transaction-broadcasting) to the peers of the Bitcoin network.

> The coordinator sends this transaction over the tor network to a random Bitcoin P2P node, and from there it is gossiped to other nodes and miners.

Author: yeah, good point... IDK if it should be removed

> The coordinator sends this transaction over the tor network to a random Bitcoin P2P node, and from there it is gossiped to other nodes and miners.

Reviewer: My point is someone might think that there is Bitcoin nodes that are not P2P.

---
{
  "title": "CoinJoin",
  "description": "A step by step guide on how to do Wasabi CoinJoins. This is the Wasabi documentation, an archive of knowledge about the open-source, non-custodial and privacy-focused Bitcoin wallet for desktop."
}
---

# CoinJoin

// What are CoinJoins?

[[toc]]

## Doing CoinJoin step-by-step

1. Launch Wasabi, and open your wallet with the coins that you want to join.

> Notice that it is not yet possible to CoinJoin from a hardware wallet, the keys must be "hot" on your computer".

Reviewer 2: ```suggestion
Notice that it is not yet possible to CoinJoin from a hardware wallet, the keys must be "hot" on your computer.
```

Then open the `CoinJoin` tab in the tab or the right side menu.

![](/CoinJoin.png)

2. Select coins for joining by clicking the check box in the coin list.
You can select one or more coins, but when several are selected, then they can be linked and clustered by an attacker.
So choose carefully which coins you want others to know belong to the same entity.
You can select red shield coins, which are not private yet, but will gain anonymity set in this round of CoinJoin.
Also you can select green shield coins, who come from a previous round of CoinJoin, so that they gain even more anonymity set.

> What is important that the sum of coins registered have a value larger than the minimum level, which is currently roughly 0.1 bitcoin.

Reviewer 2: ```suggestion
What is important is that the sum of coins registered have a value larger than the minimum level, which is currently roughly 0.1 bitcoin.
```

3. Specify anonymity set target.
Wasabi will automatically re-register your coin for the next round until this target is reached.
You can easily toggle between three levels by clicking on the shield.
You can also [change the value](/FAQ/FAQ-UseWasabi.md#how-can-i-change-the-anonset-target) of the three shields in the settings.

![](/CoinJoinAnonsetTargetYellow.png)
![](/CoinJoinAnonsetTargetGreen.png)
![](/CoinJoinAnonsetTargetCheck.png)

4. Enter password and click `Enqueue Selected Coins`.

> You only have to enter the password for the first round of mixing, the re-mix directly after is done without a password input.

Author: What I want to say here is that the automatic re-mix [the one in the exactly next round] is without the password. If the user manually re-registers a cj coin, then the password is required again.

> You only have to enter the password for the first round of mixing, the re-mix directly after is done without a password input.

Reviewer: then we shouldn't say remix here should we?
